common/db_test.go

```diff
@@ -118,6 +125,47 @@ func profilesMatch(j1 []ffmpeg.VideoProfile, j2 []ffmpeg.VideoProfile) bool {
 	return true
 }
 
+func TestDBOrchestrators(t *testing.T) {
```

Comment:
I'd like to suggest breaking this down into more succinct or "surgical" test functions, where we state more clearly the scenario and the expected outcome. IMO taking the approach of smaller functions and the specific naming paradigm nudges us to think more carefully about all the cases we want to cover. See this recent example: https://github.com/livepeer/go-livepeer/pull/655/files#diff-c20584ae20b069ed08dc6769a1c19e77.

In general I like structuring the names of test functions in 3 parts:

- The code under test (usually a function name)
- The scenario
- The expected result

While it's not necessarily how we normally name functions in Go, for the sake of readability I also like to separate these parts with an underscore.

Of course one of the downsides could be repetitive setup code, but that can be mitigated with some solid helper functions.

wdyt?

Reply:
Repetitive code would be the only downside, as you mentioned, but helper functions sound good. I like this new approach!

Reply:
😓 I'll go out on a limb here and say that I don't find the "surgical" approach to help readability or understanding the code flow for a few reasons:

* The method names are really long, which takes conscious effort to mentally parse and understand. Even if retaining the surgical approach, I'd really prefer more concise method names, and elaborating on the scenario/expected result as a comment.

* It generates a large amount of boilerplate, which we'd still have with helper functions.

* Separating each test case into its own function, we lose the benefit of code locality. It's harder to quickly scan the tests and reason about the various conditions that are covered (or might not be covered). Here, I was able to quickly see that we were missing nil/empty checks. Also, the progression here can logically be followed: testing empty case, testing one case, testing multiple cases. Splitting that up doesn't improve the readability.

* Related to code locality, one pattern I've used is to check for various conditions following the implementation. This makes it easier to scan the implementation and identify the test case that's covering it. For example, if a function checks and early-exits for conditions A, B, C, we'll have a unit-test checking for A, B, C.  The RPC tests have a bunch of these, eg
https://github.com/livepeer/go-livepeer/blob/master/server/rpc.go#L176 and https://github.com/livepeer/go-livepeer/blob/master/server/rpc_test.go#L147 (admittedly the checks here aren't quite in the proper order; that could be improved.)

Reply:
Seems like this is a harder point to achieve consensus around, since I actually find the long test functions with comments along the way harder to parse than the shorter functions I've been trying out.

Since I find consistency very important, I would love for us to land on a specific style all of us commit to.

@j0sh as the leader of this repo I believe you should drive this decision. I realize it is probably going to be long test functions, which will give me the opportunity to improve my code parsing skills :D